bin/shutter

```diff
@@ -5596,12 +5596,12 @@ sub STARTUP {
 	}
 
 	sub fct_validate_filename{
-		$filename	= shift;
-		$filename_hint	= shift;
+		my $myfilename	= shift;
+		my $myfilename_hint	= shift;
 		my @invalid_codes = (47, 92);
-		$filename->signal_connect(
+		$myfilename->signal_connect(
 			'key-press-event' => sub {
-				my $filename	= shift;
+				my $myfilename	= shift;
```

Comment:
Did you intend to name this variable the same as the one above?

Reply:
I'm not sure, how it worked originally, just renamed the variables to prevent confusion with the variables named `$filename` and `$filename_hint`  from outside of the function.

Reply:
Adding 'my' above would be enough for that

Reply:
I mean, confusion when reading the code. But I can switch back to the old variable names!

Reply:
The confusion is precisely why I left this comment. It was confusing before, not didn't become much better now. There are still 2 variables with the same name `$myfilename`. They don't conflict because this one has `my` in front of its declaration, but for human mind the names are still confusing

Reply:
Now I get you! What would you propose to make it more readable?

Reply:
I don't see how this second variable is even used. Maybe just remove it? Still need to call `shift;` though

Reply:
Like this?

Reply:
Yep, it works this way. I'm a newbie in Perl and wasn't sure if it would pick up the arguments array if called without `@_`. But you sound like it actually would, so I've removed it. Also works without it, I tested both ways now.

Reply:
Is there anything else that can be improved here?